### DISCUSSION TITLE
Multiplatform support

### DISCUSSION THREAD

**Discussion by siarhei-luskanau** [bitfireAT/dav4jvm#18] [Ideas]
Good day!

dav4jvm is a great WebDAV/CalDAV/CardDAV library for JVM.
I have a requiroment to load data from WebDAV server in Kotlin Multiplatfomt project for Android/Desktop/iOS/web targets.

We have a [PullRequest](https://github.com/bitfireAT/dav4jvm/pull/15) with Multiplatform support. This i great piece of work. But PullRequest is very big.

Let we add Multiplatform support step by step?

I'm going to replace some library to KMM compatible (Dateteme parsing, XML parser) and dav4jvm should work as befor (no functionality and public API changes).

Then let we replace OkHttpClient with Ktor client. It is public API breaking changes.

At the end let we add pultiplatform plugin to the project. In the result we will have maven artifacts for Android/Desktop/iOS/web targets.

Do you mind if I will create Pull Requests to make dav4jvm compatible with Multiplatform?

Cheers
Siarhei

**Comment by ArnyminerZ** [comment]
I think it will surely be good to support Kotlin Multiplatform, however, I also think it's quite a big change, so if it's going to be done, it surely needs to be done step by step. @rfc2822 what do you think

**Comment by ArnyminerZ** [comment]
After some internal discussion, we have decided that it's not good to do big changes on a library that works fine, we don't want to break the stability. However, we do agree that this kind of modifications might surely be useful for somebody, so we decided on keeping a list of forks that support different features. See https://github.com/bitfireAT/dav4jvm/pull/21

**Comment by siarhei-luskanau** [comment]
Good day!
I completely  agree with idea to keep library stable and don't make braking changes.

I have tried [the fork](https://github.com/McDjuady/dav4jvm) in internal project with KMM and Compose Mutliplatform. I works well on Desktop/Android/iOS targets.
But I have to add fork sources to the project. I think it will be better to publish KMM version of this library to a maven repository.

I have idea to create additional GitHub project for KMM version of dav4jvm library with Ktor instead of OkHttp.
Could you like to create a new GitHub project for KMM version with Ktor?

Siarhei
